### ISSUE TITLE
Metrics Text Endpoint Middleware Never Resets the StringReporter buffer

### ISSUE THREAD

**Issue by mthird** [alhardy/AppMetrics#48]
Calling the /metrics-text endpoint appends to the previous output in the 1.0.0-beta2 build using .NET Core.

```
***** Start - Health Checks *****


	Health Status = {status}

	PASSED CHECKS
	DEGRADED CHECKS
	FAILED CHECKS
***** End - Health Checks *****

-- End String Reporter Report: 2016-12-22T21:41:31.0203Z --

-- Start String Reporter Report: 2016-12-22T21:41:33.2383Z --

***** Start - Environment Information *****

     Assembly Name = src
  Assembly Version = 1.0.26.0
         Host Name = platform-site-2411001161-rsmc3
        Ip Address = 10.233.71.7
        Local Time = 2016-12-22T21:41:33.2406+00:00
      Machine Name = .
                OS = debian
        OS Version = 8

```

```c#
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddLogging();
            services.AddMetrics()
                .AddJsonSerialization()
                .AddHealthChecks()
                .AddMetricsMiddleware();
        }

        public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
        {
            app.UseMetrics();
        }
```

**Comment by mthird** [comment]
I believe middleware classes are singletons so the instance of StringReport will be reused for all invocations:

https://github.com/alhardy/AppMetrics/blob/dev/src/App.Metrics.Extensions.Middleware/Middleware/MetricsEndpointTextEndpointMiddleware.cs#L26

**Comment by alhardy** [comment]
@mthird Thanks for the fix, create a PR and I'll push this out with the next release. I noticed you're using Linux, curious to know how the package runs on Linux, I haven't tested that yet?

**Comment by alhardy** [comment]
Fixed directly on 1.0.0-beta3 rather than waiting on PR to release beta3.

**Comment by mthird** [comment]
Sorry, just realized the PR I created was inadvertently in my fork.  

So far, things seem to be running just fine on Linux with .NET Core 1.1.  We haven't switched to using it yet for monitoring but that should happen sometime in the next couple of weeks.
